### ISSUE TITLE
✨ `Qwant` for the search engine

### ISSUE THREAD

**Issue by neon-mmd** [neon-mmd/websurfx#317]
Provide the `qwant` search engine for the `websurfx` search engine.  To start working on this issue, you will need to create a new file with the name of the desired engine in the folder `src/engines/` which is located under the codebase (`websurfx` directory). Then in the file, just create a new struct with the name of the engine and then create two `impl` blocks for the struct. One for implementing the `SearchEngine` trait on it and the other for implementing the `parsers` to be used for aggregating results.

After that, just put the name of the desired engine in the match statement located [here,](https://github.com/neon-mmd/websurfx/blob/40b7e6d2ea7cf940d361dc1f9e1c068d19af2f9d/src/models/engine_models.rs#L144) make sure that the name of engine is in lowercase. 

After that, just move to the config file which is located at `websurfx/` directory under the code base and then just edit the config file by including the name of the search engine [here](https://github.com/neon-mmd/websurfx/blob/40b7e6d2ea7cf940d361dc1f9e1c068d19af2f9d/websurfx/config.lua#L52).  

> [!NOTE]
> The engine name to be included in the config file is `not case sensitive` but we would still recommend naming it with the camelcase naming convention.

After doing all the fore-mentioned steps, then we would suggest doing a thorough test to make sure everything works as intended and then submitting a PR.

**Comment by diogocanut** [comment]
Hey, I want to take this one if not taken : )

**Comment by neon-mmd** [comment]
> Hey, I want to take this one if not taken : )

Yes sure, we will assign this issue to you right away :rocket: :slightly_smiling_face: .

**Comment by neon-mmd** [comment]
Diogocanut, it has been nearly a month, any progress on this issue so far? We would like to know :slightly_smiling_face: .

**Comment by jfvillablanca** [comment]
if this isn't being worked on, i'd like to take this one.

**Comment by diogocanut** [comment]
@jfvillablanca you can take it, had no time to finish yet

**Comment by neon-mmd** [comment]
> @jfvillablanca you can take it, had no time to finish yet

Ok thanks :heart: diogocanut for keeping us updated. Since do not you want to work on this issue therefore we will re-assign the issue to jfvillablanca. Hope to see you soon with another issue in the project. :slightly_smiling_face:

**Comment by neon-mmd** [comment]
Jfvillablanca, we have now assigned this issue to you. You may start working ln it right away :rocket: :slightly_smiling_face: .

**Comment by jfvillablanca** [comment]
Feel free to unassign this issue from me. I can't commit some time to work on this currently.

**Comment by neon-mmd** [comment]
> Feel free to unassign this issue from me. I can't commit some time to work on this currently.

Yes sure, no problem. We will un-assign you from this issue right away :rocket:. Also, thanks :heart: for keeping us update on this issue. :slightly_smiling_face:
